csharp/src/Ice/Connection.cs

```diff
@@ -856,14 +873,14 @@ public class UdpConnection : IPConnection
         private readonly UdpSocket _udpSocket;
 
         internal UdpConnection(
-            IConnectionManager? manager,
             Endpoint endpoint,
             MultiStreamOverSingleStreamSocket socket,
-            IConnector? connector,
-            string connectionId,
+            object? label,
             ObjectAdapter? adapter)
-            : base(manager, endpoint, socket, connector, connectionId, adapter) =>
+            : base(endpoint, socket, label, adapter) =>
             _udpSocket = (UdpSocket)_socket.Underlying;
+
+        internal override bool CanTrust(NonSecure preferNonSecure) => true;
```

Comment:
I don't think that's correct. If you have a proxy with PreferNonSecure=Never, you can't use UDP "connection".

Reply:
Fixed, we now use the implementation in the base class IPConnection for both